### PR TITLE
add decrepit namespace and put SEED, IDEA, Blowfish, and CAST5 in it

### PR DESCRIPTION
**Pull request by reaperhulk**
diff is a big large but most of it is moving the `test_{seed,idea,cast5,blowfish}.py` files into `test_decrepit.py` along with lots of docs.

### CODE REVIEW

File: src/cryptography/hazmat/decrepit/ciphers/algorithms.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+from cryptography.hazmat.primitives.ciphers.algorithms import (
+    _CAST5Internal as CAST5,
+)
+from cryptography.hazmat.primitives.ciphers.algorithms import (
+    _IDEAInternal as IDEA,
+)
+from cryptography.hazmat.primitives.ciphers.algorithms import (
+    _SEEDInternal as SEED,
+)
```
Comment (alex) [inline src/cryptography/hazmat/decrepit/ciphers/algorithms.py:18]: Surely we should just do the reverse -- move the classes here, and re-export them in the other place but deprecated.

File: tests/hazmat/primitives/test_decrepit.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
+# This file is dual licensed under the terms of the Apache License, Version
```
Comment (alex) [inline tests/hazmat/primitives/test_decrepit.py:1]: This filename doesn't line up with the name of the module with teh code?

Comment (reaperhulk) [reply tests/hazmat/primitives/test_decrepit.py:1]: should we do `hazmat/primitives/decrepit/test_algorithms.py`?

Comment (alex) [reply tests/hazmat/primitives/test_decrepit.py:1]: yeah

File: src/cryptography/hazmat/decrepit/ciphers/algorithms.py (hunk 1 above)
Comment (reaperhulk) [reply src/cryptography/hazmat/decrepit/ciphers/algorithms.py:18]: Circular dependencies make this annoying, but we can do it (albeit with a significantly larger diff probably). We could also do that as a followup PR.

Comment (reaperhulk) [reply src/cryptography/hazmat/decrepit/ciphers/algorithms.py:10]: (done)

**Review by alex — APPROVED** [review]

**Comment by misterzed88** [comment]
That is a nice solution. Thank you for doing the change!

**Pull request merged**